### ISSUE TITLE
Implement gradient-based optimization in optimal_feed_profile

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#56]
## Summary

`optimal_feed_profile()` in `src/difflow/units/fed_batch.py` (line 759) is explicitly marked as a placeholder in its docstring and always returns a flat constant feed rate regardless of the requested objective:

```python
"""...
This is a placeholder for optimization-based feed profile design.
In practice, would use gradient-based optimization with difflow's
autodiff capabilities.
..."""
# Placeholder: return constant feed rate
```

The function accepts `objective` values of `"max_selectivity"`, `"min_time"`, and `"max_yield"` but ignores them entirely.

## What is Missing

Actual gradient-based optimization of the feed profile using difflow's autodiff capabilities.

## Implementation Plan

1. Parameterize the feed profile as a piecewise-constant or spline over `n_steps` control intervals
2. Define a scalar objective function per mode that simulates the fed-batch reactor via `FedBatchReactor.__call__` and returns a scalar loss:
   - `"max_selectivity"`: maximize `C_target / C_total` at end of batch
   - `"max_yield"`: maximize total moles of `target_species` produced
   - `"min_time"`: minimize time to reach a target concentration threshold
3. Use `optimistix` (already a dependency) or `jaxopt` with `jax.grad` to optimize the feed-rate parameters
4. Return a callable that interpolates the optimized piecewise profile and the optimal batch time `t_opt`
5. Add tests in `tests/test_fed_batch.py` verifying each objective mode produces a result that improves on the flat-rate baseline

## Location

`src/difflow/units/fed_batch.py`, `optimal_feed_profile`, line 759

**Comment by jkitchin** [comment]
## Fixed in commit 7d83d26

`optimal_feed_profile()` now performs real gradient-based optimization using JAX autodiff and `optimistix.BFGS`. Here's what was implemented:

### Feed profile parameterization
The feed is represented as a piecewise-constant profile over `n_intervals` (default 10) control intervals. The optimization variables are unconstrained reals that are mapped to non-negative feed rates via `softplus`.

### Volume constraint enforcement
A differentiable hard-scaling step (`jnp.minimum(1, V_budget / V_added)`) rescales the feed rates so the total volume added never exceeds `V_max - V0`. This is exact with no penalty tuning required.

### Objective functions
- **`max_yield`**: maximize final moles of `target_species`
- **`max_selectivity`**: maximize `C_target / C_total` at end of batch
- **`min_time`**: maximize time-weighted average moles of `target_species` (earlier production weighted more heavily — proxy for fast production)

### Additional fix
`FedBatchReactor.__call__` had a `TracerBoolConversionError` (`if n_initial[i] > 1e-10:`) that was only exposed when the reactor ran under `optimistix`'s JIT tracing. Replaced with `jnp.where(...)`.

### New API parameters
```python
optimal_feed_profile(
    objective,
    params, C0, T, target_species, feed_composition, V_max, t_max,
    n_intervals=10,   # control horizon (new)
    n_sim_steps=100,  # ODE integration steps (new)
)
```

### Tests added
8 new tests in `TestOptimalFeedProfile` covering all three objective modes, non-negativity of feed rates, improvement over flat-feed baseline, and volume constraint satisfaction.
